README.md

```diff
@@ -2,8 +2,72 @@
+
+### Development
+
+1. Clone the repo
+2. `cd smart-city-dashboard`
+3. `cp .env.example .env`
+4. Change the env variable `REACT_APP_API_URL` to the URL of the backend
+5. `yarn` or `npm install`
+6. `yarn start` or `npm start`
+
+#### Technology
+
 This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
 
+The frontend is written in [typescript](https://www.typescriptlang.org/) and uses [react](https://reactjs.org/). State management is done with [redux](https://redux.js.org/) and [redux-saga](https://redux-saga.js.org/). The page style and layout is done with [bulma](https://bulma.io/), while components are styled with [styled-components](https://styled-components.com/). Data visualization on the dashboard is either custom written (e.g. parking) or visualized with [apexcharts](https://apexcharts.com/) (e.g. charts).
+
+#### Dev Technology
+
+We are using typescript for type safety. In order to have consistend formatting / styling of the code, we are using [husky](https://typicode.github.io/husky/#/), [lint-staged](https://github.com/okonet/lint-staged) and [prettier](https://prettier.io/).
+
+#### File Structure
+
+- `.vscode` Settings when developing with VSCode
+- `public` Static files that will be served as is
+- `scripts` Scripts for maintaining the project
+- `src` Source code. This also includes the react entry point, service workers and types
+  - `actions` Redux actions (splitted for each data source)
+  - `components` React Components
+  - `pages` Pages of the website
+  - `reducers` Redux reducers (splitted for each data source)
+  - `resources` Assets that will be compiled (like icons, images etc)
+  - `sagas` redux sagas (Sagas are implemented using Generator functions)
+  - `views` Containers that display data (splitted for each data source)
+
+#### Available Visualizations
+
+- Tile: Displaying numbers with units and titles in tiles / cards
+- Progress: Visualizing progress which is here used to show the utilization of parking decks
+- Line diagram: visualizing (multiple) lines in a diagram
+
+#### How to add a new data source
+
+If you want to add a new data source to the dashboard, there are multiple steps neccessary:
+
+1. Start in the `src/reducers` folder and create a new reducer similar to the existing ones. Here, you will also define the state for youry particular dataset
+2. Next one, create corresponding actions for your data in the `src/actions` folder
+3. After defining how the data should be stored and managed, we can take care of fething the data. Create a new saga in the `src/sagas` folder similar to the existing files. The generator functions will fetch data regularly based on the interval you can define
+
+#### Show the data on the dashboard
+
+1. The most laborious will take place in `src/views` where you create a new container for your data. Here, you need to access the redux state and render different kinds of visualizations
```

Comment:
labour

Reply:
Resolved in [ca1fb3b](https://github.com/reedu-reengineering-education/smart-city-dashboard/pull/15/commits/ca1fb3b0457e9f148eef04857de7522da4ca8815)